src/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -45,7 +45,7 @@ public T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder)
         /// </returns>
         public IList<T> CollectionOfFake<T>(int numberOfFakes)
         {
-            var result = new List<T>();
+            var result = new SynchronizedCollection<T>();

```

Comment:
Fair enough. As long as you can guarantee in the future that nothing internal will ever use this for any reason and cast it to `IEnumerable<T>` then, no, you probably don't need that safety. I'll revert.